lib/rules/no-extra-boolean-cast.js

```diff
@@ -126,6 +129,57 @@ module.exports = {
             return Boolean(sourceCode.getCommentsInside(node).length);
         }
 
+        /**
+         * Checks if the given node is wrapped in grouping parentheses. Parentheses for constructs such as if() don't count.
+         * @param {ASTNode} node The node to check.
+         * @returns {boolean} `true` if the node is parenthesized.
+         * @private
+         */
+        function isParenthesized(node) {
+            return eslintUtils.isParenthesized(1, node, sourceCode);
+        }
+
+        /**
+         * Determines whether the given node needs to be parenthesized when replacing the previous node.
+         * @param {ASTNode} previousNode Previous node.
+         * @param {ASTNode} node The node to check.
+         * @returns {boolean} `true` if the node needs to be parenthesized.
+         */
+        function needsParens(previousNode, node) {
+            if (isParenthesized(previousNode)) {
+
+                // parentheses around the previous node will stay, so there is no need for an additional pair
+                return false;
+            }
+
+            // parent of the previous node will become parent of the replacement node
+            const parent = previousNode.parent;
+
+            switch (parent.type) {
+                case "CallExpression":
+                case "NewExpression":
+                    return node.type === "SequenceExpression";
+                case "IfStatement":
+                case "DoWhileStatement":
+                case "WhileStatement":
+                case "ForStatement":
```

Comment:
Just double checking, should ForOfStatement and ForInStatement be included here?

Reply:
No, this `switch` is supposed to cover only possible parents for the reported node in this rule (otherwise, this function could grow to the size of `no-extra-parens` rule).

It also assumes the node's role in its parent. For example, `case "ConditionalExpression"` knows that it is the `test` node at this point (precedence check would be different for `consequent` or `alternate` node).

Reply:
Gotcha, thanks. Can you add a comment explaining that?

Reply:
Added to the function's description, thanks for the suggestion!